diffprivlib/tools/histograms.py

```diff
@@ -140,12 +144,23 @@ def histogram(sample, epsilon=1.0, bins=10, range=None, weights=None, density=No
 
     hist, bin_edges = np.histogram(sample, bins=bins, range=range, weights=weights, density=None)
 
-    dp_mech = GeometricTruncated(epsilon=epsilon, sensitivity=1, lower=0, upper=maxsize, random_state=random_state)
```

Comment:
Have a look on implementation of a geometric truncated distribution using opendp.
You may check the Mechanisms section in Diffprivlib to see how this mechanism works under the hood.